src/window.ts

```diff
@@ -234,4 +250,139 @@ export class PersistedWindowLayout {
     constructor(name?: string) {
         this.name = name;
     }
-}
\ No newline at end of file
+}
+
+export class SnapAssistWindowManager {
+    private readonly container: Container;
+    private readonly floater: ContainerWindow;
+    public snapThreshold: number = 20;
+    public snapOffset: number = 15;
+    private snappingWindow: string;
+
+    public constructor(container: Container, options?: any) {
+        this.container = container;
+
+        if (options) {
+            if (options.snapThreshold || options.snapThreshold === 0) {
+                this.snapThreshold = options.snapThreshold;
+            }
+
+            if (options.snapOffset || options.snapOffset === 0) {
+                this.snapOffset = options.snapOffset;
+            }
+        }
+
+        this.attach();
+    }
+
+    public attach(win?: ContainerWindow) {
+        if (win) {
+            win.addListener("move", (e) => {
+                const id = e.sender.id;
+
+                if (this.snappingWindow === id) {
+                    return;
+                }
+
+                e.sender.getBounds().then(bounds => {
```

Comment:
As most of this is done via promises, there can be a bounce from snap to last set of coordinates raised via the move event.  https://github.com/Morgan-Stanley/desktopJS/issues/101